src/libmongoc/src/mongoc/mongoc-cursor.c

```diff
@@ -1755,6 +1771,28 @@ _mongoc_cursor_prepare_getmore_command (mongoc_cursor_t *cursor,
                          abs (_mongoc_n_return (cursor)));
    }
 
+   comment = _mongoc_cursor_get_opt (cursor, MONGOC_CURSOR_COMMENT);
+
+   if (comment != NULL && comment->value_type != BSON_TYPE_EOD) {
+      mongoc_server_stream_t *server_stream;
+
+      /* CRUD spec: If a comment is provided, drivers MUST attach this comment
+       * to all subsequent getMore commands run on the same cursor for server
+       * versions 4.4 and above. For server versions below 4.4 drivers MUST NOT
+       * attach a comment to getMore commands. */
+      server_stream = _mongoc_cursor_fetch_stream (cursor);
+
+      /* TODO: is no-op on server_stream sensible? This function has no error
+       * reporting so it's unclear can be done if we fail to fetch a stream. */
```

Comment:
Outstanding question. There really isn't a viable way to report an error here, but I believe `_mongoc_cursor_fetch_stream` already sets some error state within the cursor if it fails internally.

Presumably, a stream would already have been fetched for the cursor to decide whether to use a a getMore command or opcode, so perhaps we need not worry about an error here. If so, should we consider using an assertion on `server_stream`?

Reply:
`_mongoc_cursor_prepare_getmore_command` is called from quiet a few places, some of which don't have any error reporting, so this sounds like a more ambitious change. Do you think it's worth tracking this todo in another ticket for future work?

Reply:
@kevinAlbs: I think this is the only outstanding comment thread.

Reply:
>  Do you think it's worth tracking this todo in another ticket for future work?

IMO no. I think the likelihood it being addressed in the near future is low. If `_mongoc_cursor_fetch_stream` returns an error, I think it is likely the next call to `_mongoc_cursor_fetch_stream` will fail.